### PR TITLE
Creates Extras per Cloud provider

### PR DESCRIPTION
**Pull request by jperkelens**
As an AWS user, I do not want to install Azure modules to run snowflake and vice versa. This PR isolates each cloud provider into its own extra (GCP appears to acquire no extra modules).

### CODE REVIEW

**Comment by keller00** [comment]
I don't think doing just this is enough.
There needs to be error user friendly error messages, the connector should know if it has access to dependencies and such.
You could take a look at: https://github.com/snowflakedb/snowflake-connector-python/blob/master/options.py

**Comment by snowstakeda** [comment]
@jperkelens we are still investigating the impact of these type of changes. Hope we come up something in the next couple of months.

**Comment by jperkelens** [comment]
@keller00 happy to implement something like that!

**Comment by keller00** [comment]
Thank you, @jperkelens 
Let me know if I could be of help 😄

File: scripts/install.sh [hunk 1]
```diff
@@ -34,15 +34,15 @@ if [ "$TRAVIS_OS_NAME" == "osx" ]; then
...
 else
     if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
-         pip install .[pandas,development]
+         pip install .[pandas,development,azure,s3]
     else
         pv=${TRAVIS_PYTHON_VERSION/./}
         $THIS_DIR/build_inside_docker.sh $pv
         CONNECTOR_WHL=$(ls $THIS_DIR/../dist/docker/repaired_wheels/snowflake_connector_python*cp${PYTHON_ENV}*.whl | sort -r | head -n 1)
-        pip install -U ${CONNECTOR_WHL}[pandas,development]
+        pip install -U ${CONNECTOR_WHL}[pandas,development,azure,s3]
```
Comment (keller00) [inline scripts/install.sh]: This reminds me, `tox.ini` has to be updated too.

**Comment by keller00** [comment]
Here's someone else's PR to go off of:
https://github.com/snowflakedb/snowflake-connector-python/pull/182/files
It also reminded me about the scripts in the `scripts` directory

**Comment by jperkelens** [comment]
@keller00 alright, i believe this covers the requested changed (tho there may be some build/test/install artifacts im missing)

**Comment by keller00** [comment]
There's some more stuff missing, some tests directly import from `botocore` and others.
Did you want me to commit straight onto your branch to help out with this?

**Comment by jperkelens** [comment]
@keller00 if you don't mind, that's fine. QQ though, wouldn't that just be a matter of adding the dependency to the `development` extra then?

**Comment by keller00** [comment]
Sure that would run for now, but if we want to run testing without `aws` and `azure` extras separately then it'll be a problem.
Sorry, I'm going to be busy next week, so please don't worry I'm working on it and didn't forget about you! :)

**Pull request closed without merging**